test/error_in_creator.cpp

```diff
@@ -207,10 +207,19 @@ TEST_P(FaultyDelayedItemErrorTest, faultyCompressedItem)
   EXPECT_NO_THROW(creator.addItem(item));
   // We force the closing of the cluster, so working thread will detect error
   EXPECT_NO_THROW(creator.addMetadata("A metadata", "A compressed (default) metadata"));
-  // give a chance to threads to detect the error
-  // For compressed cluster, the cluster task is a bit longer to use the contentProvider
-  // (memory initialisation of the compression stream ?)
-  usleep(1000000);
+  // give a chance to threads to detect the error.
+  // How many time to wait is a bit tricky.
+  // Too long and all tests will wait to much and developpers hate to wait
+  // Not enough and error is not deteced and tests fail (and developpers hate failing tets)
+  // The exacte value is specific to each computer, so we need to make this configurable.
+  // We use a base and we multiply it by a factor taken from env variable.
+  long sleep_time = 1000000; // Default value is set to a factor 10 above what is needed to work on my (fast) computer
+  char* str_time_factor = std::getenv("WAIT_TIME_FACTOR_TEST");
+  if (str_time_factor) {
+    // Yes, if user set a "non int" value, sleep_time will be 0. But not our problem here
+    sleep_time *= atoi(str_time_factor);
+  }
```

Comment:
Duplication with the other unit test (`FaultyDelayedItemErrorTest.faultyUnCompressedItem`). You can get rid of it by introducing a helper function `double getWaitTimeFactor()` (and I meant `double` so developers hating to wait have an option to lower the wait time).